components/workouts/ExerciseList.jsx

```diff
@@ -0,0 +1,154 @@
+/* eslint-disable react/forbid-prop-types */
+import * as React from 'react';
+import { SafeAreaView, View, TouchableOpacity } from 'react-native';
+import styled from 'styled-components/native';
+import AlphabetSectionList from 'react-native-alphabet-sectionlist';
+import { SearchBar } from 'react-native-elements';
+import ExerciseItem from './ExerciseItem';
+
+const Title = styled.Text`
+  font-family: 'Montserrat_600SemiBold';
+  font-size: 24px;
+  margin: 15px 5%;
+`;
+
+const SectionHeader = styled.Text`
+  font-family: 'Montserrat_600SemiBold';
+  font-size: 20px;
+  padding-left: 2%;
+  background-color: #CAB0FF;
+  color: #EFEFEF;
+
+`;
+
+const ButtonContainer = styled(TouchableOpacity)`
+   background-color: #E2E2E2;
+   border-radius: 20px;
+   padding: 5px;
+   padding-left: 15px;
+   padding-right: 15px;
+   position: absolute;
+   top: 14px;
+   right: 20px;
+`;
+
+const Buttontext = styled.Text`
+  font-family: 'Montserrat_500Medium';
+  font-size: 18px;
+`;
+
+const parseItems = (items) => {
+  // Sort names alphabetically
+  items.sort((a, b) => a.name.localeCompare(b.name));
+
+  // Group by first letter of each name
+  const bucketData = items.reduce((accumulator, item) => {
+    const bucket = item.name[0].toUpperCase();
+
+    // If this is the first time we've seen this letter, create a bucket
+    if (!accumulator[bucket]) {
+      accumulator[bucket] = [item];
+    } else {
+      accumulator[bucket].push(item);
+    }
+
+    return accumulator;
+  }, {});
+  return bucketData;
+};
+
+const renderHeader = ({ section }) => (
+  <SectionHeader>{section.title}</SectionHeader>
+);
+
+const ExerciseList = (props) => {
+  const { items } = props;
+
+  const parsedItems = parseItems(items);
+  const [search, setSearch] = React.useState('');
+  const [filteredDataSource, setFilteredDataSource] = React.useState([]);
+  const [masterDataSource, setMasterDataSource] = React.useState([]);
+  React.useEffect(() => {
+    setFilteredDataSource(parsedItems);
+    setMasterDataSource(parsedItems);
+  }, []);
```

Comment:
If the only purpose of this useEffect is to set the initial state you can do this, problem with the way you're doing it is that you're passing it an array initially which is not what the data prop in alphabetSectionList expects so we get a warning and sometimes it'll act weird
```suggestion
  const [filteredDataSource, setFilteredDataSource] = React.useState(parsedItems);
  const [masterDataSource, setMasterDataSource] = React.useState(parsedItems);
```